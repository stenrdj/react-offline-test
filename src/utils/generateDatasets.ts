import { GraphTileDataType, fuel } from "../types/GraphTile.type";
import { FuelTypeColor } from "../types/LineGraph";

const getDatasetTypeColor = (type: fuel) => {
  return FuelTypeColor[type];
};

export const getLabelsFromData = (data: GraphTileDataType) => {
  return [data.from];
};

export const getDatasetFromData = (data: GraphTileDataType) => {
  const datasets = data?.generationmix?.map((type) => {
    return {
      label: type.fuel.toUpperCase(),
      data: [type.perc],
      backgroundColor: getDatasetTypeColor(type.fuel),
    };
  });
  return datasets;
};
