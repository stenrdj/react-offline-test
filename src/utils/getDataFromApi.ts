import { GraphTileDataType } from "../types/GraphTile.type";

export const getEnergyGeneration = () =>
  fetch("https://api.carbonintensity.org.uk/generation")
    .then((response) => response.json())
    .then((response) => response.data);
