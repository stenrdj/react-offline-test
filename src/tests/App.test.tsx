import React from "react";
import renderer from "react-test-renderer";
import App from "../app";
import { data } from "./utils.test";

it("app renders", async () => {
  const component = await renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
