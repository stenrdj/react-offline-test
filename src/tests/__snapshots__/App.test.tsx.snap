// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1oqqzyl-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;padding-left:16px;padding-right:16px;}@media (min-width:600px){.css-1oqqzyl-MuiContainer-root{padding-left:24px;padding-right:24px;}}@media (min-width:1200px){.css-1oqqzyl-MuiContainer-root{max-width:1200px;}}",
      }
    }
    data-emotion="css 1oqqzyl-MuiContainer-root"
  />,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-u6x4pz{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;-webkit-align-items:baseline;-webkit-box-align:baseline;-ms-flex-align:baseline;align-items:baseline;padding:1em;border:1px solid #ccc;border-radius:2px;}",
        }
      }
      data-emotion="css u6x4pz"
    />
    <div
      className="MuiBox-root css-u6x4pz"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-45zogx-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400px;font-size:1rem;line-height:1.5px;letter-spacing:0.00938em;}",
          }
        }
        data-emotion="css 45zogx-MuiTypography-root"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 css-45zogx-MuiTypography-root"
      >
        UK Energy Mix
      </p>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-ap8bhd-MuiButton-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500px;font-size:0.875rem;line-height:1.75px;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;}.css-ap8bhd-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-ap8bhd-MuiButton-root:hover{background-color:transparent;}}.css-ap8bhd-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
          }
        }
        data-emotion="css ap8bhd-MuiButton-root"
      />
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-18rz1ss-MuiButtonBase-root-MuiButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500px;font-size:0.875rem;line-height:1.75px;letter-spacing:0.02857em;text-transform:uppercase;min-width:64px;padding:5px 15px;border-radius:4px;-webkit-transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border:1px solid rgba(25, 118, 210, 0.5);color:#1976d2;}.css-18rz1ss-MuiButtonBase-root-MuiButton-root::-moz-focus-inner{border-style:none;}.css-18rz1ss-MuiButtonBase-root-MuiButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-18rz1ss-MuiButtonBase-root-MuiButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-18rz1ss-MuiButtonBase-root-MuiButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(25, 118, 210, 0.04);border:1px solid #1976d2;}@media (hover: none){.css-18rz1ss-MuiButtonBase-root-MuiButton-root:hover{background-color:transparent;}}.css-18rz1ss-MuiButtonBase-root-MuiButton-root.Mui-disabled{color:rgba(0, 0, 0, 0.26);border:1px solid rgba(0, 0, 0, 0.12);}",
          }
        }
        data-emotion="css 18rz1ss-MuiButtonBase-root-MuiButton-root"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-18rz1ss-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Update
      </button>
    </div>
  </div>,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-4iw138-MuiContainer-root{width:100%;margin-left:auto;box-sizing:border-box;margin-right:auto;display:block;padding-left:16px;padding-right:16px;padding-top:2em;}@media (min-width:600px){.css-4iw138-MuiContainer-root{padding-left:24px;padding-right:24px;}}@media (min-width:1200px){.css-4iw138-MuiContainer-root{max-width:1200px;}}",
      }
    }
    data-emotion="css 4iw138-MuiContainer-root"
  />,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-4iw138-MuiContainer-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-k008qs{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        }
      }
      data-emotion="css k008qs"
    />
    <div
      className="MuiBox-root css-k008qs"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-gosxm2{border:1px solid #ccc;padding:1em;border-radius:2px;width:calc(50% - 24px);display:inline-block;margin-bottom:24px;}",
          }
        }
        data-emotion="css gosxm2"
      />
      <div
        className="MuiBox-root css-gosxm2"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-tgwjxk-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500px;font-size:1.25rem;line-height:1.6px;letter-spacing:0.0075em;margin-bottom:1em;}",
            }
          }
          data-emotion="css tgwjxk-MuiTypography-root"
        />
        <h6
          className="MuiTypography-root MuiTypography-h6 css-tgwjxk-MuiTypography-root"
        >
          Energy overview for dates below :
        </h6>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1l4w6pd{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
            }
          }
          data-emotion="css 1l4w6pd"
        />
        <div
          className="MuiBox-root css-1l4w6pd"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1vbn3lh-MuiTextField-root{margin-right:10px;}.css-1vbn3lh-MuiTextField-root .clearButton{opacity:1px;}@media (pointer: fine){.css-1vbn3lh-MuiTextField-root .clearButton{opacity:0;}.css-1vbn3lh-MuiTextField-root:hover .clearButton,.css-1vbn3lh-MuiTextField-root:focus-within .clearButton{opacity:1px;}}",
              }
            }
            data-emotion="css 1vbn3lh-MuiTextField-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-b22k0j-MuiFormControl-root-MuiTextField-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:relative;min-width:0;padding:0;margin:0;border:0;vertical-align:top;margin-right:10px;}.css-b22k0j-MuiFormControl-root-MuiTextField-root .clearButton{opacity:1px;}@media (pointer: fine){.css-b22k0j-MuiFormControl-root-MuiTextField-root .clearButton{opacity:0;}.css-b22k0j-MuiFormControl-root-MuiTextField-root:hover .clearButton,.css-b22k0j-MuiFormControl-root-MuiTextField-root:focus-within .clearButton{opacity:1px;}}",
              }
            }
            data-emotion="css b22k0j-MuiFormControl-root-MuiTextField-root"
          />
          <div
            className="MuiFormControl-root MuiTextField-root css-b22k0j-MuiFormControl-root-MuiTextField-root"
            onClick={[Function]}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": "@-webkit-keyframes mui-auto-fill{from{display:block;}}@keyframes mui-auto-fill{from{display:block;}}@-webkit-keyframes mui-auto-fill-cancel{from{display:block;}}@keyframes mui-auto-fill-cancel{from{display:block;}}",
                }
              }
              data-emotion="css-global 1prfaxn"
            />
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400px;font-size:1rem;line-height:1.4375em;letter-spacing:0.00938em;color:rgba(0, 0, 0, 0.87);box-sizing:border-box;position:relative;cursor:text;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;border-radius:4px;padding-right:14px;}.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root.Mui-disabled{color:rgba(0, 0, 0, 0.38);cursor:default;}.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.87);}@media (hover: none){.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root:hover .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.23);}}.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline{border-color:#1976d2;border-width:2px;}.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root.Mui-error .MuiOutlinedInput-notchedOutline{border-color:#d32f2f;}.css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root.Mui-disabled .MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.26);}",
                }
              }
              data-emotion="css p6ujsk-MuiInputBase-root-MuiOutlinedInput-root"
            />
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input{font:inherit;letter-spacing:inherit;color:currentColor;padding:4px 0 5px;border:0;box-sizing:content-box;background:none;height:1.4375em;margin:0;-webkit-tap-highlight-color:transparent;display:block;min-width:0;width:100%;-webkit-animation-name:mui-auto-fill-cancel;animation-name:mui-auto-fill-cancel;-webkit-animation-duration:10ms;animation-duration:10ms;padding:16.5px 14px;padding-right:0;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-webkit-input-placeholder{color:currentColor;opacity:0.42px;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-moz-placeholder{color:currentColor;opacity:0.42px;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:-ms-input-placeholder{color:currentColor;opacity:0.42px;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-ms-input-placeholder{color:currentColor;opacity:0.42px;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:focus{outline:0;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:invalid{box-shadow:none;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-webkit-search-decoration{-webkit-appearance:none;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-webkit-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-moz-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:-ms-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input::-ms-input-placeholder{opacity:0!important;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:focus::-webkit-input-placeholder{opacity:0.42px;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:focus::-moz-placeholder{opacity:0.42px;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:focus:-ms-input-placeholder{opacity:0.42px;}label[data-shrink=false]+.MuiInputBase-formControl .css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:focus::-ms-input-placeholder{opacity:0.42px;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input.Mui-disabled{opacity:1px;-webkit-text-fill-color:rgba(0, 0, 0, 0.38);}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:-webkit-autofill{-webkit-animation-duration:5000s;animation-duration:5000s;-webkit-animation-name:mui-auto-fill;animation-name:mui-auto-fill;}.css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input:-webkit-autofill{border-radius:inherit;}",
                  }
                }
                data-emotion="css 1fl9bx-MuiInputBase-input-MuiOutlinedInput-input"
              />
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                inputMode="text"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onPaste={[Function]}
                placeholder="MM/DD/YYYY hh:mm aa"
                readOnly={false}
                required={false}
                type="text"
                value="03/26/2024 02:50 AM"
              />
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-1laqsz7-MuiInputAdornment-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:0.01em;max-height:2em;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;white-space:nowrap;color:rgba(0, 0, 0, 0.54);margin-left:8px;}",
                  }
                }
                data-emotion="css 1laqsz7-MuiInputAdornment-root"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-do3kej-MuiIconButton-root{text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;margin-right:-12px;}.css-do3kej-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-do3kej-MuiIconButton-root:hover{background-color:transparent;}}.css-do3kej-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
                    }
                  }
                  data-emotion="css do3kej-MuiIconButton-root"
                />
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;margin-right:-12px;}.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root::-moz-focus-inner{border-style:none;}.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root:hover{background-color:transparent;}}.css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
                    }
                  }
                  data-emotion="css 1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                />
                <button
                  aria-label="Choose date, selected date is Mar 26, 2024"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <style
                    dangerouslySetInnerHTML={
                      {
                        "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
                      }
                    }
                    data-emotion="css i4bv87-MuiSvgIcon-root"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                  </svg>
                </button>
              </div>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-rki20w-MuiOutlinedInput-notchedOutline{border-color:rgba(0, 0, 0, 0.23);}",
                  }
                }
                data-emotion="css rki20w-MuiOutlinedInput-notchedOutline"
              />
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-1d3z3hw-MuiOutlinedInput-notchedOutline{text-align:left;position:absolute;bottom:0;right:0;top:-5px;left:0;margin:0;padding:0 8px;pointer-events:none;border-radius:inherit;border-style:solid;border-width:1px;overflow:hidden;min-width:0%;border-color:rgba(0, 0, 0, 0.23);}",
                  }
                }
                data-emotion="css 1d3z3hw-MuiOutlinedInput-notchedOutline"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-ihdtdm{float:unset;width:auto;overflow:hidden;padding:0;line-height:11px;-webkit-transition:width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;transition:width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;}",
                    }
                  }
                  data-emotion="css ihdtdm"
                />
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-slev9r-MuiPickersPopper-root{z-index:1300px;}",
              }
            }
            data-emotion="css slev9r-MuiPickersPopper-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1egwj5j-MuiPopper-root-MuiPickersPopper-root{z-index:1300px;}",
              }
            }
            data-emotion="css 1egwj5j-MuiPopper-root-MuiPickersPopper-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1phu4zx-MuiTextField-root .clearButton{opacity:1px;}@media (pointer: fine){.css-1phu4zx-MuiTextField-root .clearButton{opacity:0;}.css-1phu4zx-MuiTextField-root:hover .clearButton,.css-1phu4zx-MuiTextField-root:focus-within .clearButton{opacity:1px;}}",
              }
            }
            data-emotion="css 1phu4zx-MuiTextField-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1kxm7oe-MuiFormControl-root-MuiTextField-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;position:relative;min-width:0;padding:0;margin:0;border:0;vertical-align:top;}.css-1kxm7oe-MuiFormControl-root-MuiTextField-root .clearButton{opacity:1px;}@media (pointer: fine){.css-1kxm7oe-MuiFormControl-root-MuiTextField-root .clearButton{opacity:0;}.css-1kxm7oe-MuiFormControl-root-MuiTextField-root:hover .clearButton,.css-1kxm7oe-MuiFormControl-root-MuiTextField-root:focus-within .clearButton{opacity:1px;}}",
              }
            }
            data-emotion="css 1kxm7oe-MuiFormControl-root-MuiTextField-root"
          />
          <div
            className="MuiFormControl-root MuiTextField-root css-1kxm7oe-MuiFormControl-root-MuiTextField-root"
            onClick={[Function]}
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": "@-webkit-keyframes mui-auto-fill{from{display:block;}}@keyframes mui-auto-fill{from{display:block;}}@-webkit-keyframes mui-auto-fill-cancel{from{display:block;}}@keyframes mui-auto-fill-cancel{from{display:block;}}",
                }
              }
              data-emotion="css-global 1prfaxn"
            />
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-p6ujsk-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1fl9bx-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r2:"
                inputMode="text"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onPaste={[Function]}
                placeholder="MM/DD/YYYY hh:mm aa"
                readOnly={false}
                required={false}
                type="text"
                value="03/26/2024 02:50 AM"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="Choose date, selected date is Mar 26, 2024"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-ihdtdm"
                >
                  <span
                    className="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <p>
          Loading...
        </p>
      </div>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-xwwgja{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:50%;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;}",
          }
        }
        data-emotion="css xwwgja"
      />
      <div
        className="MuiBox-root css-xwwgja"
      />
    </div>
  </div>,
]
`;
